hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Server.java

```diff
@@ -1325,7 +1325,7 @@ public String toString() {
   }
 
   /** Listens on the socket. Creates jobs for the handler threads*/
-  private class Listener extends Thread {
+  protected class Listener extends Thread {
```

Comment:
Would have been better if we could keep this private, but it seems we have no better way if we want to test this change (Thanks for the test btw). If we are going to go forward with this, we should annotate this as `@VisibleForTesting` and `@InterfaceAudience.Private`.

Let's also get some expert advice on this. cc @aajisaka @ayushtkn @jojochuang